src/model/topology.cpp

```diff
@@ -691,6 +692,12 @@ void Topology::Spec(const Topology::Specs& specs)
       }
       read_fill_network = it->second;
 
+      if(typeid(*read_fill_network) == typeid(LegacyNetwork)) {
```

Comment:
I tried adding a virtual function for GetSpecs but it seems I couldn't override the virtual func with inherited return type. 
```
In file included from include/model/topology.hpp:44,
                 from include/model/engine.hpp:37,
                 from include/mapspaces/mapspace-base.hpp:33,
                 from src/mapspaces/mapspace-base.cpp:28:
include/model/network-legacy.hpp:194:10: error: ‘model::LegacyNetwork::Specs& model::LegacyNetwork::GetSpecs()’ marked ‘override’, but does not override
  194 |   Specs& GetSpecs() override { return specs_; }
      |          ^~~~~~~~
compilation terminated due to -fmax-errors=1.
scons: *** [build/mapspaces/mapspace-base.o] Error 1
```

It is a bit weird since `Specs` is inherited from `NetworkSpecs` and based on StackOverflow it is allowed
I will FillLatency() and DrainLatency() instead for now.

Reply:
I believe that will only work if you return a *pointer* to NetworkSpecs instead of a reference. But that would mean changing the code for all callers of that function to work with the pointer instead of a reference. And then you'll need the xxxLatency() virtual functions at the NetworkSpecs class (and all its child classes). Too many changes.

I think what you're proposing to do is fine, i.e., directly provide the xxxLatency() functions at the Network class for now.

Reply:
Yes, returning a pointer to the NetworkSpecs seems to work, but it incurs another error from the boost serialization library which I couldn't find a solution (it seems to be a linker issue). 

```
/usr/bin/ld: build/applications/mapper/mapper.o: in function `boost::serialization::extended_type_info_typeid<mod
el::Network>::extended_type_info_typeid()':
/usr/include/boost/serialization/extended_type_info_typeid.hpp:91: undefined reference to `typeinfo for model::Ne
twork'
/usr/bin/ld: build/libtimeloop-model.a(network-legacy.o): in function `model::Network::~Network()':
/root/timeloop-1/include/model/network.hpp:84: undefined reference to `vtable for model::Network'
/usr/bin/ld: build/libtimeloop-model.a(network-legacy.o): in function `model::Network::Network()':
/root/timeloop-1/include/model/network.hpp:81: undefined reference to `vtable for model::Network'
/usr/bin/ld: build/libtimeloop-model.a(network-legacy.o): in function `model::Network::Network(model::Network con
st&)':
/root/timeloop-1/include/model/network.hpp:81: undefined reference to `vtable for model::Network'
/usr/bin/ld: build/libtimeloop-model.a(network-legacy.o):(.data.rel.ro._ZTIN5model13LegacyNetworkE[_ZTIN5model13L
egacyNetworkE]+0x10): undefined reference to `typeinfo for model::Network'
/usr/bin/ld: build/libtimeloop-model.a(network-legacy.o): in function `model::LegacyNetwork const* boost::seriali
zation::smart_cast_impl::pointer<model::LegacyNetwork const*>::polymorphic::cast<model::Network const>(model::Net
work const*)':
/usr/include/boost/serialization/smart_cast.hpp:184: undefined reference to `typeinfo for model::Network'
/usr/bin/ld: build/libtimeloop-model.a(network-reduction-tree.o):(.data.rel.ro._ZTVN5model20ReductionTreeNetworkE
[_ZTVN5model20ReductionTreeNetworkE]+0x88): undefined reference to `model::Network::GetSpecsPtr()'
/usr/bin/ld: build/libtimeloop-model.a(network-reduction-tree.o):(.data.rel.ro._ZTIN5model20ReductionTreeNetworkE
[_ZTIN5model20ReductionTreeNetworkE]+0x10): undefined reference to `typeinfo for model::Network'
/usr/bin/ld: build/libtimeloop-model.a(network-reduction-tree.o): in function `model::ReductionTreeNetwork const*
 boost::serialization::smart_cast_impl::pointer<model::ReductionTreeNetwork const*>::polymorphic::cast<model::Net
work const>(model::Network const*)':
/usr/include/boost/serialization/smart_cast.hpp:184: undefined reference to `typeinfo for model::Network'
/usr/bin/ld: build/libtimeloop-model.a(network-simple-multicast.o):(.data.rel.ro._ZTVN5model22SimpleMulticastNetw
orkE[_ZTVN5model22SimpleMulticastNetworkE]+0x88): undefined reference to `model::SimpleMulticastNetwork::GetSpecs
Ptr()'
```